Compile/src/Producers/PrefiringWeightProducer.cc

```diff
@@ -1,162 +1,338 @@
+    parametrization0p83To1p24_ = (TF1*) file_prefiringparams_->Get(paramName);
+    paramName = "L1prefiring_muonparam_1.24To1.4_" + dataeraMuon_;
+    parametrization1p24To1p4_ = (TF1*) file_prefiringparams_->Get(paramName);
+    paramName = "L1prefiring_muonparam_1.4To1.6_" + dataeraMuon_;
+    parametrization1p4To1p6_ = (TF1*) file_prefiringparams_->Get(paramName);
+    paramName = "L1prefiring_muonparam_1.6To1.8_" + dataeraMuon_;
+    parametrization1p6To1p8_ = (TF1*) file_prefiringparams_->Get(paramName);
+    paramName = "L1prefiring_muonparam_1.8To2.1_" + dataeraMuon_;
+    parametrization1p8To2p1_ = (TF1*) file_prefiringparams_->Get(paramName);
+    paramName = "L1prefiring_muonparam_2.1To2.25_" + dataeraMuon_;
+    parametrization2p1To2p25_ = (TF1*) file_prefiringparams_->Get(paramName);
+    paramName = "L1prefiring_muonparam_2.25To2.4_" + dataeraMuon_;
+    parametrization2p25To2p4_ = (TF1*) file_prefiringparams_->Get(paramName);
 
-    double prefrate = h_prefmap->GetBinContent(thebin);
+    if (parametrization0p0To0p2_ == nullptr || parametrization0p2To0p3_ == nullptr ||
+        parametrization0p3To0p55_ == nullptr || parametrization0p55To0p83_ == nullptr ||
+        parametrization0p83To1p24_ == nullptr || parametrization1p24To1p4_ == nullptr ||
+        parametrization1p4To1p6_ == nullptr || parametrization1p6To1p8_ == nullptr ||
+        parametrization1p8To2p1_ == nullptr || parametrization2p1To2p25_ == nullptr ||
+        parametrization2p25To2p4_ == nullptr) {
+        missingInputMuon_ = true;
+        LOG(ERROR) << this->GetProducerId() << ": "
+            << "Muon parametrization not found for at least one bin. All prefiring weights set to 0.";
+    }
 
-    double statuncty = h_prefmap->GetBinError(thebin);
-    double systuncty = prefiringRateSystUnc_ * prefrate;
+    paramName = "L1prefiring_muonparam_HotSpot_" + dataeraMuon_;
+    parametrizationHotSpot_ = (TF1*) file_prefiringparams_->Get(paramName);
+    file_prefiringparams_->Close();
+    if ((dataeraMuon_.find("2016") != std::string::npos) && parametrizationHotSpot_ == nullptr) {
+        missingInputMuon_ = true;
+        LOG(ERROR) << this->GetProducerId() << ": "
+            << "Year is 2016 and no Muon parametrization is found for hot spot. All prefiring weights set to 0.";
+    }
 
-    if (fluctuation == up)
-        prefrate = std::min(1., prefrate + sqrt(pow(statuncty, 2) + pow(systuncty, 2)));
-    if (fluctuation == down)
-        prefrate = std::max(0., prefrate - sqrt(pow(statuncty, 2) + pow(systuncty, 2)));
-    return prefrate;
 }
 
-
 void PrefiringWeightProducer::Produce(ZJetEvent const& event,
                                  ZJetProduct& product,
                                  ZJetSettings const& settings) const
 {
-    double nonPrefiringProba[3] = {1., 1., 1.};  //0: central, 1: up, 2: down
+    //Photons
+    const auto& thePhotons = product.m_pfPhotons;
+
+    //Jets
+    const auto& theJets = product.m_correctedZJets["L1L2L3Res"];
```

Comment:
Could another correction level be necessary at some point?

Reply:
It says fully corrected PAT Jets. So ye this could change if the full correction level changes. But for now prefiring is only in Run II and I think the Run II Jet correction levels won't change.
But I could make this a config key.

Reply:
Okay than. Since this is the official recipe, I don't think it is necessary.